pandas/_typing.py

```diff
@@ -141,7 +141,10 @@
 ]
 
 # For functions like rename that convert one label to another
-Renamer = Union[Mapping[Hashable, Any], Callable[[Hashable], Hashable]]
```

Comment:
Is there a reason to have the mapping return Any while the function returns Hashable?

Reply:
I think it should be `Hashable`, because we want hashable objects as names of `index` and `columns` .  It's what I have in my work-in-progress on the MS type stubs.

We want the type checkers to catch this as an error:
```python
s7 = pd.Series([1, 2, 3]).rename({1: [3, 4, 5]})
```

If you use `Hashable` instead of `Any`, then `mypy` and `pyright` will pick it up